src/charm.py

```diff
@@ -213,6 +278,142 @@ def _on_get_rule_action(self, event: ActionEvent) -> None:
         event.log(f"Successfully fetched rule: {rule_id}")
         event.set_results(rule)
 
+    def _on_auth_proxy_config_changed(self, event: AuthProxyConfigChangedEvent) -> None:
+        if not self._oathkeeper_service_is_running:
+            self.unit.status = WaitingStatus("Waiting for Oathkeeper service")
+            event.defer()
+            return
+
+        if not self._peers:
+            self.unit.status = WaitingStatus("Waiting for peer relation")
+            event.defer()
+            return
+
+        access_rules_locations = list()
+        for rule_type in ("allow", "deny"):
+            rules = self._render_access_rules(
+                rule_type=rule_type,
+                protected_urls=event.protected_urls,
+                allowed_endpoints=event.allowed_endpoints,
+                relation_app_name=event.relation_app_name,
+            )
+
+            if rules:
+                # Push the rules to the container
+                filename = f"{self._oathkeeper_access_rules_path}/access-rules-{event.relation_app_name}-{rule_type}.json"
+                self._container.push(filename, str(rules), make_dirs=True)
+                access_rules_locations.append(filename)
+
+        self._set_auth_proxy_relation_peer_data(
+            event.relation_id, dict(access_rules_locations=access_rules_locations)
+        )
+
+        self._push_oathkeeper_config()
```

Comment:
If pushing the file fails, the application will believe that it is protected, but it won't be. I think we should first push the config and then set the peer data.

Reply:
I'm adding the relation's access rules locations to peer data first so that the config gets updated with existing peer data + new dirs rather than overwritten. I put it in a try-except block instead. Is it ok for you?